resources/2024-02-output-resources.md

```diff
@@ -0,0 +1,210 @@
+# Adding ProvisioningStatus field to the OutputResource
+
+* **Status**: In Progress
+* **Author**: Nithya Subramanian (@nithyatsu)
+
+## Overview
+
+Output resources are the actual platform specific resources that get deployed in order to deploy a Radius resource. For example to deploy a container as part of radius application deployment on kubernetes, we deploy a kubernetes ServiceAccount, Role, RoleBinding and Deployment Output resources.
+
+There are a couple of problems related to how we manage output resources today -
+
+1. Output resources do not have a ProvisioningStatus property. Instead, if any of the output resource deployment fails, we set the Provisioning state of the corresponding radius resource accordingly and do not store the output resources. With this approach, we lose the information on which output resources failed deployment since we aggregate failures into the owning Radius resource's provisioningStatus property. 
+
+2. We do not record any of the output resources as part of the radius resource whose deployment failed, although we deployed them with an errored status. This leads to orphaned resources which should be manually cleaned up.
+
+This design proposes addition of ProvisioningStatus to outputResources so that, 
+once we add changes to track these resources in case of both successful and failed deployment to avoid orphaning resources, there will be clearer indication of what lead to failed deployment to the user. 
```

Comment:
I want to push back on one aspect of this. The `operationStatus` and `operationResult` are the things that describe the result of an operation. The resource itself describes the state of the resource. They have different purposes.

I'm saying this because we need to keep the purpose in mind when we're designing. For the example the error messages or other debugging information are not part of the resource, they are part of the `operationStatus`, and it should stay that way.

Reply:
I am ok with adding the provisioningState for output resources, but we should be careful how we're applying this reasoning.